### PR TITLE
Keptn plugin added

### PR DESCRIPTION
**Pull request by huseyinbabal**
## Description

Changes proposed in this pull request:

- Added Keptn plugin

Screenshot 2023-07-20 at 15 05 47

## Testing
You can refer to [this article](https://botkube.io/blog/implementing-your-own-botkube-plugin-a-real-life-use-case) for testing. In that article I am explaining also howto implement plugin, but you don't need to implement it at all, only test.

## Related issue(s)

#1089

### CODE REVIEW

File: internal/source/keptn/client.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+type Data struct {
+	Message string
+	Project string
+	Service string
+	Status  string
+	Stage   string
+	Result  string
+}
```
Comment (pkosiec) [inline internal/source/keptn/client.go:35]: Please add comments to the structures. Thanks!

File: internal/source/keptn/client.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	}, api.EventsGetEventsOptions{})
+	if err != nil {
+		return nil, err.ToError()
+	}
+
+	for _, ev := range res {
+		data := Data{}
+		err := ev.DataAs(&data)
+		if err != nil {
+			return nil, err
```
Comment (pkosiec) [inline internal/source/keptn/client.go:74]: pleasae wrap the error

File: internal/source/keptn/source.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+		JSONSchema:  jsonSchema(),
+	}, nil
+}
+
+func (p *Source) consumeEvents(ctx context.Context, cfg Config, ch chan<- source.Event) {
+	keptn, err := NewClient(cfg.URL, cfg.Token)
+	log := loggerx.New(cfg.Log)
+	exitOnError(err, log)
+
+	for {
```
Comment (pkosiec) [inline internal/source/keptn/source.go:63]: I think we should exit the for loop in case `ctx.Done()`. Please update this. Thank you!

**Review by pkosiec — COMMENTED** [review]
Just a code review, testing this plugin now 👍

File: internal/source/keptn/source.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+			  "project": {
+				"description": "Keptn Project",
+				"type": "string"
+			  },
+			  "service": {
+				"description": "Keptn Service",
+				"type": "string"
+			  }
+			},
+			"required": []
```
[the remark this answers is not in the document]
Comment (huseyinbabal) [reply internal/source/keptn/source.go:138]: only token is mandatory, I will send update accordingly

File: internal/source/keptn/client.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	res, err := c.API.Events().GetEvents(ctx, &filter, api.EventsGetEventsOptions{})
+	if err != nil {
+		return nil, err.ToError()
+	}
+
+	for _, ev := range res {
+		data := Data{}
+		err := ev.DataAs(&data)
+		if err != nil {
+			return nil, fmt.Errorf("while mapping Keptn event to internal event %w", err)
```
Comment (pkosiec) [inline internal/source/keptn/client.go:86]:
```suggestion
			return nil, fmt.Errorf("while mapping Keptn event to internal event: %w", err)
```

File: internal/source/keptn/source.go [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+		select {
+		case <-ticker.C:
+			req := GetEventsRequest{
+				Project:  cfg.Project,
+				Service:  cfg.Service,
+				FromTime: time.Now().Add(-time.Second * pollPeriodInSeconds),
+			}
+			res, err := keptn.Events(ctx, &req)
+			if err != nil {
+				log.Errorf("failed to get events. %v", err)
```
Comment (pkosiec) [inline internal/source/keptn/source.go:74]:
```suggestion
				log.Errorf("while getting events: %v", err)
```

**Review by pkosiec — APPROVED** [review]
LGTM, last small comments 👍

**Pull request merged**